### PR TITLE
refactor(gui-v2): Smartsheet toolbar

### PR DESCRIPTION
**Pull request by pranavxc**
## Change Summary

- Fields menu
    - Sortable
    - Filter
    - System fields
- Filter menu 
    - Grouped filter 
- Sort menu
- Share view
- Share view list
- More options
- Table reload icon
- Table delete icon
- Lock type menu
- Csv download

re #2818

## Change type

- [x] feat: (new feature for the user, not a new feature for build script)
- [ ] fix: (bug fix for the user, not a fix to a build script)
- [ ] docs: (changes to the documentation)
- [ ] style: (formatting, missing semi colons, etc; no production code change)
- [ ] refactor: (refactoring production code, eg. renaming a variable)
- [ ] test: (adding missing tests, refactoring tests; no production code change)
- [ ] chore: (updating grunt tasks etc; no production code change)

## Test/ Verification

Screenshot 2022-07-29 at 5 56 59 PM
Screenshot 2022-07-29 at 5 57 17 PM
Screenshot 2022-07-29 at 5 57 26 PM
Screenshot 2022-07-29 at 5 57 35 PM
Screenshot 2022-07-29 at 5 57 47 PM
Screenshot 2022-07-29 at 5 57 58 PM
Screenshot 2022-07-29 at 5 58 08 PM
Screenshot 2022-07-29 at 5 58 23 PM

### CODE REVIEW

**Comment by dstala** [comment]
Verification [root DB: SQLite, project DB: MySQL]
- [x] Show / hide system fields
- [x] Show all / hide all
- [x] Fields re-order
- [ ] Filter operation, abbreviations listed. Expand.
  
Screenshot 2022-07-30 at 2 02 39 PM
- [ ] Filter didn't work (auto-apply enabled/disabled). WIP?
  
Screenshot 2022-07-30 at 2 04 04 PM
- [ ] Filter field select icons not clear, alignment missing for text. v1 & v2 images pasted below
  
Screenshot 2022-07-30 at 2 06 41 PM
Screenshot 2022-07-30 at 2 06 07 PM

UI
- [ ] Font size for `fields` menu,  `show all`, `hide all` can be reduced by 1 point

**Comment by dstala** [comment]
0730 1900
- [ ] decrease default column width for spreadsheet ID (#) : mark it close to 4-5 characters.
- [ ] filter with `auto apply` disabled doesn't work even after explicitly clicking `apply filter` button
- [ ] sort isn't working always. works once in a few attempts. looked pretty random
- [ ] share view: increase window width to ensure entire link comes in a single line
- [ ] share view access broken. wip I believe.
- [ ] more options menu persists over share view modal
  
Screenshot 2022-07-30 at 7 26 14 PM

File: packages/nc-gui-v2/components/smartsheet-toolbar/SearchData.vue [hunk 1]
```diff
@@ -27,9 +16,26 @@ const columns = computed(() =>
 </script>
 
 <template>
-  <a-input v-model:value="localValue" size="small" class="max-w-[250px]" placeholder="Filter query">
+  <a-input
+    v-model:value="search.value"
```
Comment (bcakmakoglu) [inline packages/nc-gui-v2/components/smartsheet-toolbar/SearchData.vue:20]: v-model should be on `search` not `search.value`

File: packages/nc-gui-v2/components/smartsheet-toolbar/ShareView.vue [hunk 2]
```diff
@@ -1,13 +1,165 @@
...
+  } catch (e) {
+    toast.error(await extractSdkResponseErrorMsg(e))
+  }
+
+  $e('a:view:share:enable-pwd')
+}
+
+const copyLink = () => {
+  copy(sharedViewUrl?.value as string)
+  toast.info('Copied to clipboard')
```
Comment (bcakmakoglu) [inline packages/nc-gui-v2/components/smartsheet-toolbar/ShareView.vue:98]: maybe use a [`message`](https://antdv.com/components/message) for this? @dstala What do you say?
Toast seems strange for the clipboard message

File: packages/nc-gui-v2/components/smartsheet/Toolbar.vue [hunk 3]
```diff
@@ -13,11 +13,24 @@
...
+    <!--    <SmartsheetToolbarLockMenu />
+    <div class="dot" />
+    <SmartsheetToolbarReload />
+    <div class="dot" />
+    <SmartsheetToolbarAddRow />
+    <div class="dot" />
+    <SmartsheetToolbarDeleteTable />
+    <div class="dot" />
+    <SmartsheetToolbarToggleDrawer class="mr-2" />
+ -->
```
Comment (bcakmakoglu) [inline packages/nc-gui-v2/components/smartsheet/Toolbar.vue:25]: can be removed

File: packages/nc-gui-v2/components/smartsheet/Toolbar.vue [hunk 4]
```diff
@@ -13,11 +13,24 @@
...
 </template>
 
 <style scoped>
 :deep(.nc-toolbar-btn) {
   @apply border-0 !text-xs font-semibold px-2;
-}
+} /*
+.dot {
+  @apply w-[3px] h-[3px] bg-gray-300 mx-1 rounded-full;
+}*/
```
Comment (bcakmakoglu) [inline packages/nc-gui-v2/components/smartsheet/Toolbar.vue:35]: can be removed

File: packages/nc-gui-v2/components/tabs/Smartsheet.vue [hunk 5]
```diff
@@ -1,11 +1,12 @@
 <script setup lang="ts">
-import type { ColumnType } from 'nocodb-sdk'
+import type { ColumnType, TableType, ViewType } from 'nocodb-sdk'
 import { ViewTypes } from 'nocodb-sdk'
+import type { Ref } from 'vue'
 import SmartsheetGrid from '../smartsheet/Grid.vue'
 import { computed, inject, provide, useMetas, watch, watchEffect } from '#imports'
+import { useProvideSmartsheetStore } from '~/composables/useSmartsheetStore'
```
Comment (bcakmakoglu) [inline packages/nc-gui-v2/components/tabs/Smartsheet.vue:7]: move `useProvideSmartsheetStore` to import from '#imports'

File: packages/nc-gui-v2/composables/useDashboard.ts [hunk 6]
```diff
@@ -0,0 +1,10 @@
+export default () => {
```
Comment (bcakmakoglu) [inline packages/nc-gui-v2/composables/useDashboard.ts:1]: Change to `export function useDashboard() {}`

**Review by bcakmakoglu — CHANGES_REQUESTED** [review]

File: packages/nc-gui-v2/components/smartsheet-toolbar/ShareView.vue (hunk 2 above)
Comment (dstala) [reply packages/nc-gui-v2/components/smartsheet-toolbar/ShareView.vue:98]: > 

what would be the difference between these two? if `message`, where will it be displayed?

File: packages/nc-gui-v2/components/smartsheet-toolbar/SearchData.vue (hunk 1 above)
Comment (pranavxc) [reply packages/nc-gui-v2/components/smartsheet-toolbar/SearchData.vue:20]: In the above case `search` is a `reactive({ value: '' })` variable and `value` is a property in object.

File: packages/nc-gui-v2/components/smartsheet-toolbar/ShareView.vue (hunk 2 above)
Comment (pranavxc) [reply packages/nc-gui-v2/components/smartsheet-toolbar/ShareView.vue:98]: Message looks neat, will update with that

File: packages/nc-gui-v2/composables/useDeleteTable.ts [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+export function useDeleteTable() {
+  const { $e, $api } = useNuxtApp()
+  const toast = useToast()
+  const { getMeta, removeMeta } = useMetas()
+  const { loadTables } = useProject()
+  const { closeTab } = useTabs()
+
+  const deleteTable = (table: TableType) => {
+    $e('c:table:delete')
+    // 'Click Submit to Delete The table'
```
[the remark this answers is not in the document]
Comment (pranavxc) [reply packages/nc-gui-v2/composables/useDeleteTable.ts]: Merged into a single composable

**Review by bcakmakoglu — APPROVED** [review]

**Review by dstala — APPROVED** [review]
minor corrections will be dealt with post merge

**Pull request merged**